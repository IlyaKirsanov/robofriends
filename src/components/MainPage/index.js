import React from 'react'
//import axios from 'axios'
import CardList from '../CardList'
import SearchBox from '../SearcBox'
import ErrorBoundaries from '../ErrorBoundary'
import Scroll from '../Scroll'
import CounterButton from '../CounterButton'
import './MainPage.css'

class MainPage extends React.Component {

	componentDidMount() {
		this.props.onRequestRobots();
	}

	filteredRobots = robots => {
		return robots.filter(robot => {
			return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase())
		})
	}

	render() {
		const {onSearchChange, robots, isPending } = this.props;


		let robotsList = null;

		if (!isPending) {
			robotsList = <h2>Loading Data ... </h2>
		} else {
			robotsList = <CardList robots={this.filteredRobots(robots)} />
		}

		return (
			<div className="tc">
				<h1>RoboFriends</h1>
				<SearchBox searchChange={onSearchChange} />
				<CounterButton color="cyan" />
				<hr style={{ backgroundColor: 'cyan', border: 'none', height: '1px' }} />
				<Scroll>
					<ErrorBoundaries>
						{robotsList}
					</ErrorBoundaries>
				</Scroll>
			</div>
		)
	}
}


export default (MainPage);