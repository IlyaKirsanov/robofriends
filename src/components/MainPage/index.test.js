import React from 'react'
import MainPage from './index.js'
import { shallow } from 'enzyme'


let wrapper;
beforeEach(() => {
	const mockProps = {
		onRequestRobots: jest.fn(),
		robots: [],
		searchField: '',
		isPending: false
	}

	wrapper = shallow(<MainPage {...mockProps}/>)
})

it('render MainPage without crashing', ()=>{
	expect(wrapper).toMatchSnapshot()
})

it('filters robots correctly', ()=>{

	const mockProps2 = {
		onRequestRobots: jest.fn(),
		robots: [
			{id: 3, name: 'John', email: 'john@test.com'}
		],
		searchField: 'john',
		isPending: false
	}

	
	expect(wrapper.instance().filteredRobots([])).toEqual([])

	let wrapper2 = shallow(<MainPage {...mockProps2} />)
	expect(wrapper.instance().filteredRobots([])).toEqual([])
})