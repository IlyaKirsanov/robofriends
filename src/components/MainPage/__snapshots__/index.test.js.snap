// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MainPage without crashing 1`] = `
<div
  className="tc"
>
  <h1>
    RoboFriends
  </h1>
  <SearchBox />
  <CounterButton
    color="cyan"
  />
  <hr
    style={
      Object {
        "backgroundColor": "cyan",
        "border": "none",
        "height": "1px",
      }
    }
  />
  <Scroll>
    <ErrorBoundary>
      <h2>
        Loading Data ... 
      </h2>
    </ErrorBoundary>
  </Scroll>
</div>
`;
